### ISSUE TITLE
Graduate The `ScrapeConfig` CRD To `v1beta1`

### ISSUE THREAD

**Issue by mviswanathsai** [prometheus-operator/prometheus-operator#6697]
### What is missing? Please describe.

To improve adoption among the users and progress with the CRD, we need to graduate the `ScrapeConfig` CRD to `v1beta1`. 

A good place to start would be to add a design proposal outlining how we will be extending the `ScrapeConfig` API surface and eventually graduating it from `v1alpha1` to `v1beta1`.

## Environment
Kubernetes Version:
Prometheus-Operator Version:

**Comment by simonpasquier** [comment]
We need all tasks from https://github.com/prometheus-operator/prometheus-operator/milestone/11 to be complete before considering v1beta1.

**Comment by slashpai** [comment]
@mviswanathsai Looking at the [proposal(https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/proposals/202405-scrapeconfig-graduation.md), IIUC we have covered all the SDs mentioned. So currently we are doing refactoring for existing ones right?

**Comment by mviswanathsai** [comment]
That's right, we need to refactor still. I haven't gotten around to it yet, figured it might be doable as GFIs for new contributors. But if we are required to do it for a release, I am up for working on it again.

**Comment by slashpai** [comment]
I think we can log the issues for SDs that need refactoring so that we someone can pick that up or I can also pitch in if you need help 🙂
